lighthouse-viewer/app/src/lighthouse-report-viewer.js

```diff
@@ -163,6 +178,10 @@ class LighthouseReportViewer {
       let saveCallback = null;
       if (!this._reportIsFromGist) {
         saveCallback = this._onSaveJson;
+      }
+
+      // Only modify history if not from a gist and not using PSI.
```

Comment:
the `"clear gist from query string"` from above is referring to this, so should be moved/removed (I wish this behavior had some tests).

Maybe update the comment to be more like `// Clear the query string if current report isn't from a gist or PSI.` ?